### ISSUE TITLE
[BUG]: Long waiting times before job start working on premise agents (up to two minutes)

### ISSUE THREAD

**Issue by beddead** [microsoft/azure-pipelines-agent#4272]
### What happened?

We see very long delays before agents starts execute jobs after update from 2.217.2 to 3.220.0
What agent is waiting so long before starting?

See time 10:19:29Z in agent log

[Agent_20230514-085616-utc.log](https://github.com/microsoft/azure-pipelines-agent/files/11471771/Agent_20230514-085616-utc.log)
[Worker_20230514-101929-utc.log](https://github.com/microsoft/azure-pipelines-agent/files/11471772/Worker_20230514-10
[logs_137498.zip](https://github.com/microsoft/azure-pipelines-agent/files/11471773/logs_137498.zip)
1929-utc.log)

### Versions

Agents: 2.320.0
OS: Linux 5.10.179-std-def-alt1 ALT Server 10.1 (Mendelevium) ALT 
OS: Linux 5.15.0-33-generic Astra Linux CE 2.12.45 (Orel)

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

Azure DevOps Server (Please specify exact version in the textbox below)

### Azure DevOps Server Version (if applicable)

Azure DevOps Server 2022.0.1

### Operation system

Linux x64

### Version controll system

git

### Relevant log output

```shell
[2023-05-14 10:19:29Z INFO JobNotification] Entering JobStarted Notification
[2023-05-14 10:19:29Z INFO JobNotification] Entering StartMonitor
[2023-05-14 10:20:29Z INFO JobDispatcher] Successfully renew job request 240251, job is valid till 14.05.2023 10:30:28
[2023-05-14 10:21:29Z INFO JobDispatcher] Successfully renew job request 240251, job is valid till 14.05.2023 10:31:28
[2023-05-14 10:21:43Z INFO ProcessInvokerWrapper] STDOUT/STDERR stream read finished.
[2023-05-14 10:21:43Z INFO ProcessInvokerWrapper] STDOUT/STDERR stream read finished.
[2023-05-14 10:21:43Z INFO ProcessInvokerWrapper] Exited process 5158 with exit code 100
[2023-05-14 10:21:43Z INFO ProcessInvokerWrapper] Finished process 5158 with exit code 100, and elapsed time 00:02:14.3692682.
[2023-05-14 10:21:43Z INFO JobDispatcher] Worker finished for job 39e15aa1-bfd9-5ab3-d587-71538ea08f55. Code: 100
[2023-05-14 10:21:43Z INFO JobDispatcher] finish job request for job 39e15aa1-bfd9-5ab3-d587-71538ea08f55 with result: Succeeded
[2023-05-14 10:21:43Z INFO Terminal] WRITE LINE: 2023-05-14 10:21:43Z: задание Сборка rpm-пакета выполнено с результатом: Succeeded
```

**Comment by HumanPrinter** [comment]
I think this is a duplicate of https://github.com/microsoft/azure-pipelines-agent/issues/4215

**Comment by LiliaSabitova** [comment]
Hi @beddead thank you for reporting!
This should be fixed with the next version of agent.

**Comment by ageisen2000** [comment]
@LiliaSabitova  when is the next version being released?

**Comment by beddead** [comment]
Fixed in agent version 3.220.2
